opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/measurement_consumer.py

```diff
@@ -123,4 +123,6 @@ def collect(
                 for measurement in measurements:
                     metric_reader_storage.consume_measurement(measurement)
 
-        return self._reader_storages[metric_reader].collect()
+            result = self._reader_storages[metric_reader].collect()
+
+        return result
```

Comment:
```suggestion
            return metric_reader_storage.collect()
```

This change of where the lock is seems fine to me

Reply:
Weird, I remember applying it. NVM, trying again to apply.